src/helpers/queryDataParser.js

```diff
@@ -3,14 +3,29 @@ import _shuffle from 'lodash/shuffle';
 
 import { consts, texts } from '../config';
 import { QUERY_TYPES } from '../queries';
-import { eventDate } from './dateTimeHelper';
+import { GenericType } from '../types';
+import { eventDate, isBeforeEndOfToday, isTodayOrLater } from './dateTimeHelper';
+import { getGenericItemDetailTitle, getGenericItemRootRouteName } from './genericTypeHelper';
 import { mainImageOfMediaContents } from './imageHelper';
 import { momentFormat } from './momentHelper';
 import { shareMessage } from './shareHelper';
 import { subtitle } from './textHelper';
 
 const { ROOT_ROUTE_NAMES } = consts;
 
+const filterGenericItems = (item) => {
+  if (item?.genericType === GenericType.Job) {
+    const dateEnd = item?.dates?.[0]?.dateEnd;
+    const hasNotEnded = dateEnd ? isTodayOrLater(dateEnd) : true;
+
+    const publicationDate = item?.publicationDate;
+    const isPublished = publicationDate ? isBeforeEndOfToday(publicationDate) : true;
+
+    return hasNotEnded && isPublished;
+  }
+  return true;
+};
```

Comment:
nice, we can use this, until the server is able to return the already filtered data

Reply:
✅